components/centraldashboard-angular/backend/app/api.ts

```diff
@@ -20,8 +21,77 @@ export class Api {
   constructor(
       private k8sService: KubernetesService,
       private metricsService?: MetricsService,
+      private workgroupApi?: WorkgroupApi,
     ) {}
 
+  /**
+   * Middleware to check if user has access to a specific namespace.
+   * Users can access a namespace if they:
+   * - Contain any role binding within the namespace (owner, contributor, or viewer)
+   * - Are a cluster admin
+   * - Are in basic auth mode (non-identity aware clusters)
+   */
+  private async checkNamespaceAccess(request: Request, response: Response, next: NextFunction) {
+    const namespace = request.params.namespace;
+    if (!namespace) {
+      return apiError({
+        res: response,
+        code: 400,
+        error: 'Namespace parameter is required',
+      });
+    }
+
+    // If no workgroup API is configured, allow access (backward compatibility)
+    if (!this.workgroupApi) {
+      return next();
+    }
+
+    // If no user is attached to request, deny access
+    if (!request.user) {
+      return apiError({
+        res: response,
+        code: 401,
+        error: 'Authentication required to access namespace activities',
+      });
+    }
+
+    try {
+      // For non-authenticated users in basic auth mode, allow access
+      if (!request.user.hasAuth) {
+        return next();
+      }
+
+      // Get user's workgroup information
+      const workgroupInfo = await this.workgroupApi.getWorkgroupInfo(request.user);
+
+      // Check if user is cluster admin
+      if (workgroupInfo.isClusterAdmin) {
+        return next();
+      }
+
+      // Check if user has access to the specific namespace
+      const hasAccess = workgroupInfo.namespaces.some(
+        binding => binding.namespace === namespace
+      );
+
+      if (!hasAccess) {
+        return apiError({
+          res: response,
+          code: 403,
+          error: `Access denied. You do not have permission to view activities for namespace '${namespace}'.`,
+        });
+      }
+
+      next();
+    } catch (err) {
+      console.error('Error checking namespace access:', err);
```

Comment:
Please use proper long pronounceable and expressive variable names
```suggestion
    } catch (error) {
      console.error('Error checking namespace access:', error);
```